homeassistant/components/roon/media_player.py

```diff
@@ -359,11 +387,25 @@ def mute_volume(self, mute=True):
 
     def volume_up(self) -> None:
         """Send new volume_level to device."""
-        self._server.roonapi.change_volume_percent(self.output_id, 3)
+        if self._attr_volume_fixed:
+            _LOGGER.error("Cannot change volume for %s, volume is fixed", self.name)
+            return
+
+        if self._attr_volume_incremental:
+            self._server.roonapi.change_volume_raw(self.output_id, 1, "relative_step")
+        else:
+            self._server.roonapi.change_volume_percent(self.output_id, 3)
 
     def volume_down(self) -> None:
         """Send new volume_level to device."""
-        self._server.roonapi.change_volume_percent(self.output_id, -3)
+        if self._attr_volume_fixed:
+            _LOGGER.error("Cannot change volume for %s, volume is fixed", self.name)
+            return
```

Comment:
This check should be unnecessary, `volume_down` won't be called for an entity which does not set `MediaPlayerEntityFeature.VOLUME_SET` or `MediaPlayerEntityFeature.VOLUME_STEP`